tools/codegen/src/generator/languages/c/builder.rs

```diff
@@ -110,8 +110,7 @@ impl GenBuilder for ast::Union {
             let name = if default.is_byte() {
                 "NULL".to_owned()
             } else {
-                let name = default.default_constant();
-                format!("&{}", name)
+                default.default_constant()
```

Comment:
The C code of `default.default_constant()` is something like 
```C
MOLECULE_API_DECORATOR const uint8_t MolDefault_Word[2] = {____, ____};
```
the `mol_union_builder_initialize`'s 4th argument accept `const uint8_t *`, so, it's OK to pass `MolDefault_World` (type: `const uint8_t *`) to `mol_union_builder_initialize`;

---------------
In history commit, we pass `&MolDefault_UnionA` (type: `const uint8_t (*)[2]`) to `mol_union_builder_initialize`, so, it won't compile successfully.

Reply:
This PR is ready for review.
@yangby-cryptape @driftluo @code-monad